tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java

```diff
@@ -52,4 +63,46 @@ protected void assertAdditionalMessageProperties(final Message msg) {
 
         assertThat(msg.isDurable()).isTrue();
     }
+
+    /**
+     * Checks that device QoS level is ignored for events.
+     *
+     * @param ctx The test context.
+     *
+     * @throws InterruptedException if the test fails.
+     */
+    @Test
+    public void testDeviceQosLevelIsIgnored(final VertxTestContext ctx) throws InterruptedException {
```

Comment:
I had imagined to add some assertions to the existing integration tests that we have for uploading telemetry/event messages to adapters. On the consumer side we would simply need to assert that the consumed messages contain the correct value for the *qos* application property, right?

Reply:
I took a look on how this could possibly be implemented.

In Hono's integration tests base classes there is a method called _assertAdditionalMessageProperties_. As far as I can see this could be the right place for verifying the qos application property. 
For this an additional abstract method, say _getExpectedQoS_, could be defined in the base classes and overridden in the test classes.

For this approach to work each test subclass needs to focus on testing with a defined qos (as the abstract method can only be implemented once per subclass), like done in the MQTT Adapter tests: there is a subclass _TelemetryMqttQoS1IT_ and _TelemetryMqttQoS0IT_.

Consequently I would need to refactor test classes of all adapters so that they follow the pattern in the MQTT to put this to work.

WDYT?

Reply:
I can also imagine adding an *assertQosLevel(Message, int)* method to the base test classes which is then invoked in addition to the *assertAdditionalMessageProperties* method. We would then pass the expected QoS level into the generic *testUploadMessages* method(s) where the downstream consumer is created. We would then not need to refactor all test classes ...

Reply:
Turned out I couldn't implemented strictly following either my or your proposal, since the test cases the different adapters are implement following different approaches.

I managed to find something a regard as a minimal as possible change by fitting the approach to the test case at hand.